DeepRegNet/DeepReg
# #397: Add entry-point for extra data downloads

Similarly to deepreg_train/predict/warp/etc. this provides a simple command line tool for users to download the data, config and demos folders to their computer without needing to checkout the git repo (aka, those who just `pip install deepreg`).

# Description

Usage:

`deepreg_download  --output_dir XXX`
or
`deepreg_download -d XXX`

Not using the directory output will download into the current working directory.

Testing this:

1. Clone from github, delete the ./config, ./demos and ./data folder.
2. Create your anaconda env as in install (`conda env create -f environment.yml;
conda activate deepreg` or `conda env create -f environment_cpu.yml;
conda activate deepreg` if CPU only)
3. `pip install -e .` from within the repo's main directory
4. Enter either of the usage options as above
5. Should download all files from the three folders needed (config, data, demos)

You can view updated docs here: https://deepreg--553.org.readthedocs.build/en/553/getting_started/install.html

Before adding tests I wanted to make sure this works for others - so if someone can please test and let me know.

Fixes #397 

## Type of change

What types of changes does your code introduce to DeepReg?

_Please check the boxes that apply after submitting the pull request._

- [x] Documentation Update (fix or improvement on the documentation)

## Checklist

_Please check the boxes that apply after submitting the pull request._

_If you're unsure about any of them, don't hesitate to ask. We're here to help! This is
simply a reminder of what we are going to look for before merging your code._

- [x] I have [installed pre-commit](https://deepreg.readthedocs.io/en/latest/contributing/setup.html) using `pre-commit install` and formatted all changed files. If you are not certain, run `pre-commit run --all-files`.
- [x] My commits' message styles matches [our requested structure] https://deepreg.readthedocs.io/en/latest/contributing/commit.html), e.g. `Issue #<issue number>: detailed message`.
- [x] I have updated the [change log file](https://github.com/DeepRegNet/DeepReg/blob/main/CHANGELOG.md) regarding my changes.
